components/brave_wallet/browser/sns_resolver_task.cc

```diff
@@ -313,9 +313,7 @@ std::optional<std::string> ParseAndVerifyTextRecordData(
   if (record_item.version == SnsRecordsVersion::kRecordsV1) {
     // https://bonfida.github.io/solana-name-service-guide/registry.html
     // Parse NameRegistry data as a string trimming possible zeros at the end.
-    return std::string(
-        std::string(sol_record_payload.begin(), sol_record_payload.end())
-            .c_str());
+    return std::string(base::as_string_view(sol_record_payload));
```

Comment:
the comment explicitly states "Parse NameRegistry data as a string trimming possible zeros at the end".

I believe after this change the trim does not happen anymore.

Not sure if this is critical, but worth a closer look.